x-pack/apm-server/aggregation/spanmetrics/aggregator_test.go

```diff
@@ -484,10 +542,25 @@ func TestAggregatorOverflow(t *testing.T) {
 	err = agg.ProcessBatch(context.Background(), &batch)
 	require.NoError(t, err)
 	assert.Empty(t, batchMetricsets(t, batch))
+	assert.Equal(t, 1, observed.FilterMessage("service destination groups reached 50% capacity").Len())
+	assert.Len(t, observed.TakeAll(), 1)
+
+	// After hitting 50% capacity (two buckets), then subsequent new metrics will
+	// be aggregated without span.name.
+	span3 := makeSpan("service", "agent", "destination3", "trg_type_3", "trg_name_3", "success", 100*time.Millisecond, 1)
+	span4 := makeSpan("service", "agent", "destination4", "trg_type_4", "trg_name_4", "success", 100*time.Millisecond, 1)
+	span4.Span.Name = span3.Span.Name
```

Comment:
Why are we setting the span4.name to the span3.name? It is not obvious why this is being done. From what I can read in the test:

1. Add 2 distinct span with different span names (`service:destination1`, `service:destination2`). Assert we have reached 50% capacity, no metric sets need to be published immediately (2/4).
2. Add 2 more spans with the same name?  (`service:destination3`, `service:destination3`). Assert we don't get any spans to publish immediately, but we're in fact aggregating the two spans above in the same key since we don't take the `Span.Name` into consideration.

Wouldn't it be better to remove this statement? `Span.Name` won't be considered, so why bother setting it to anything.

Reply:
I don't know what I was thinking now. I've removed that line, leaving `Span.Name` as the distinct generated values.